kraft/commands/devel/dependency.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# Authors: Gaulthier Gain <gaulthier.gain@uliege.be>
+#
+# Copyright (c) 2020, NEC Europe Ltd., NEC Corporation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import sys
+
+import click
+
+from kraft.commands.list import update
+from kraft.components.library import Library
+from kraft.components.types import RepositoryType
+from kraft.context import kraft_context
+from kraft.errors import KraftError
+from kraft.logger import logger
+from kraft.utils import op, dir
+
+import os
+import subprocess
+
+TOOLCHAIN_PATH = 'tools'
+TOOLCHAIN_URL_GIT = 'https://github.com/gaulthiergain/tools.git'
```

Comment:
I'm wondering whether the tools that are referenced in this repo should not be merged with kraft itself and provided as individual executables, e.g. `kraft-devel-depend`.  The man `kraft` executable could simply proxy to this program via the subcommand: `kraft devel depend`?

Reply:
Good point, we can provide only the binary. But as for the kraft repo, given the source code allows to have new contributors.

Reply:
I think it's still feasible to integrate both sources into the same directory, since they're effectively part of "the same tool" -- one that is designed to aid in building unikraft unikernels.